codec_specs.md

```diff
@@ -14,6 +14,46 @@ Important Note:
 
 The intention behind this list is `NOT` to list all existing audio and video codecs, but rather to list those codecs that are `currently supported` in Matroska and therefore need a well defined codec ID so that all developers supporting Matroska will use the same ID. If you feel we missed support for a very important codec, please tell us on our development mailing list (cellar at ietf.org).
 
+## Defining Matroska Codec Support
+
+Support for a codec is defined in Matroska with the following values.
+
+### Codec ID
+
+Each codec supported for storage in Matroska MUST have a unique Codec ID. Each Codec ID MUST be prefixed with a two character string according to the associated type of the codec as listed in the following table.
+
+Codec Type | Codec ID Prefix
+-----------|----------------
+Video      | V_
+Audio      | A_
+Subtitle   | S_
+Button     | B_
+
+Each Codec ID MUST then use a `Major Codec ID` and may optionally use a `Minor Codec ID` and `Micro Codec ID` with a forward-slash (`/`) delimiter between each component.
```

Comment:
Unfortunately this clashes a bit with existing practice from way back when Matroska was very young. We used the following monstrosities for AAC:

* `A_AAC/MPEG2/LC`
* `A_AAC/MPEG2/MAIN`
* `A_AAC/MPEG2/LC/SBR`
* `A_AAC/MPEG2/SSR`
* `A_AAC/MPEG4/LC`
* `A_AAC/MPEG4/LTP`
* `A_AAC/MPEG4/MAIN`
* `A_AAC/MPEG4/LC/SBR`
* `A_AAC/MPEG4/SSR`

As you can see some of them have one more layer. No `CodecPrivate` was used with them.

However, those IDs have been discontinued in favor of `A_AAC` with `CodecPrivate` 2005-12-07 (mkvmerge 1.6.5).

All other codec IDs that mkvmerge handles keep to major/minor/micro.

Looking forward I'm in favor of restricting official IDs to major/minor/micro.